src/libs/common/ide/fake/FakeIDE.ts

```diff
@@ -17,7 +19,7 @@ export default class FakeIDE implements IDE {
   clipboard: FakeClipboard;
   private disposables: Disposable[] = [];
 
-  constructor() {
+  constructor(private original?: IDE) {
```

Comment:
Not a fan of this setup.  I would prefer a normalized IDE in this PR.  I think it is in scope because you are being forced to break the `FakeIDE` abstraction.  Lmk if you want me to do that for you; I don't mind either way

Reply:
If you have a good idea for a normalized ide, please go ahead.